File: dart_sample/lib/demos/material/stepper_demo_page.dart
import 'package:flutter/material.dart';

class StepperDemoPage extends StatefulWidget {
  const StepperDemoPage({super.key});

  @override
  State<StepperDemoPage> createState() => _StepperDemoPageState();
}

class _StepperDemoPageState extends State<StepperDemoPage> {
  int _currentStep = 0;
  bool _horizontal = false;
  bool _expanded = false;

  @override
  Widget build(BuildContext context) {
    final ColorScheme colors = Theme.of(context).colorScheme;
    return Column(
      crossAxisAlignment: CrossAxisAlignment.stretch,
      spacing: 10,
      children: <Widget>[
        const Text(
          'ExpandIcon + Stepper',
          style: TextStyle(fontSize: 20, color: Colors.black),
        ),
        const Text(
          'Flutter-shaped disclosure animation plus vertical/horizontal step progress, states, connectors, and controls.',
          style: TextStyle(fontSize: 14, color: Colors.black54),
        ),
        Row(
          mainAxisSize: MainAxisSize.min,
          spacing: 8,
          children: <Widget>[
            ExpandIcon(
              isExpanded: _expanded,
              onPressed: (_) => setState(() => _expanded = !_expanded),
              expandedColor: const Color(0xFF6750A4),
            ),
            Text(_expanded ? 'Expanded' : 'Collapsed'),
            const ExpandIcon(onPressed: null),
            const Text('Disabled'),
            TextButton(
              onPressed: () => setState(() => _horizontal = !_horizontal),
              child: Text(_horizontal ? 'Use vertical' : 'Use horizontal'),
            ),
          ],
        ),
        Text(
          'Current step: ${_currentStep + 1} / 3',
          style: const TextStyle(fontSize: 13, color: Color(0xFF006C4C)),
        ),
        Expanded(
          child: Stepper(
            type: _horizontal ? StepperType.horizontal : StepperType.vertical,
            currentStep: _currentStep,
            onStepTapped: (int index) => setState(() => _currentStep = index),
            onStepContinue: () =>
                setState(() => _currentStep = (_currentStep + 1).clamp(0, 2)),
            onStepCancel: () =>
                setState(() => _currentStep = (_currentStep - 1).clamp(0, 2)),
            connectorColor: WidgetStateProperty.resolveWith<Color>(
              (Set<WidgetState> states) => states.contains(WidgetState.selected)
                  ? colors.primary
                  : colors.outlineVariant,
            ),
            connectorThickness: 2,
            headerPadding: const EdgeInsetsDirectional.symmetric(
              horizontal: 20,
            ),
            contentPadding: const EdgeInsetsDirectional.only(
              start: 56,
              end: 20,
              bottom: 20,
            ),
            steps: _buildSteps(colors),
          ),
        ),
      ],
    );
  }

  List<Step> _buildSteps(ColorScheme colors) => <Step>[
    _buildStep(
      0,
      'Account',
      'Choose an account',
      'Account settings are ready.',
      colors,
    ),
    _buildStep(
      1,
      'Details',
      'Review preferences',
      'Notification and sync preferences.',
      colors,
    ),
    _buildStep(
      2,
      'Confirm',
      'Finish setup',
      'Everything is ready to submit.',
      colors,
    ),
  ];

  Step _buildStep(
    int index,
    String title,
    String subtitle,
    String content,
    ColorScheme colors,
  ) {
    return Step(
      title: Text(title),
      subtitle: Text(subtitle),
      label: Text('${index + 1}'),
      content: Container(
        padding: const EdgeInsets.all(12),
        color: const Color(0xFFF3EDF7),
        child: Text(content),
      ),
      state: index < _currentStep
          ? StepState.complete
          : index == _currentStep
          ? StepState.editing
          : StepState.indexed,
      isActive: index <= _currentStep,
      stepStyle: index == 2
          ? StepStyle(
              border: Border.all(color: colors.outline),
              gradient: LinearGradient(
                colors: <Color>[
                  colors.secondaryContainer,
                  colors.primaryContainer,
                ],
              ),
            )
          : null,
    );
  }
}

File: src/Sample/Plumix.Sample/Demos/Material/StepperDemoPage.cs
using Avalonia;
using Avalonia.Media;
using System;
using System.Collections.Generic;
using Plumix.Material;
using Plumix.Rendering;
using Plumix.UI;
using Plumix.Widgets;

namespace Plumix;

// Dart parity source: dart_sample/lib/demos/material/stepper_demo_page.dart
public sealed class StepperDemoPage : StatefulWidget
{
    public override State CreateState() => new StepperDemoPageState();
}

internal sealed class StepperDemoPageState : State
{
    private int _currentStep;
    private bool _horizontal;
    private bool _expanded;

    public override Widget Build(BuildContext context)
    {
        ColorScheme colors = Theme.Of(context).ColorScheme;
        return new Column(
            crossAxisAlignment: CrossAxisAlignment.Stretch,
            spacing: 10,
            children:
            [
                new Text("ExpandIcon + Stepper", fontSize: 20, color: Colors.Black),
                new Text(
                    "Flutter-shaped disclosure animation plus vertical/horizontal step progress, states, connectors, and controls.",
                    fontSize: 14,
                    color: Color.Parse("#8A000000")),
                new Row(
                    mainAxisSize: MainAxisSize.Min,
                    spacing: 8,
                    children:
                    [
                        new ExpandIcon(
                            isExpanded: _expanded,
                            onPressed: _ => SetState(() => _expanded = !_expanded),
                            expandedColor: Color.Parse("#FF6750A4")),
                        new Text(_expanded ? "Expanded" : "Collapsed", fontSize: 14),
                        new ExpandIcon(onPressed: null),
                        new Text("Disabled", fontSize: 14),
                        new TextButton(
                            child: new Text(_horizontal ? "Use vertical" : "Use horizontal"),
                            onPressed: () => SetState(() => _horizontal = !_horizontal)),
                    ]),
                new Text($"Current step: {_currentStep + 1} / 3", fontSize: 13, color: Color.Parse("#FF006C4C")),
                new Expanded(new Stepper(
                    type: _horizontal ? StepperType.Horizontal : StepperType.Vertical,
                    currentStep: _currentStep,
                    onStepTapped: index => SetState(() => _currentStep = index),
                    onStepContinue: () => SetState(() => _currentStep = Math.Min(2, _currentStep + 1)),
                    onStepCancel: () => SetState(() => _currentStep = Math.Max(0, _currentStep - 1)),
                    connectorColor: WidgetStateProperty<Color>.ResolveWith(states =>
                        states.Contains(WidgetState.Selected) ? colors.Primary : colors.OutlineVariant),
                    connectorThickness: 2,
                    headerPadding: EdgeInsetsGeometry.DirectionalOnly(start: 20, end: 20),
                    contentPadding: EdgeInsetsGeometry.DirectionalOnly(start: 56, end: 20, bottom: 20),
                    steps: BuildSteps(colors))),
            ]);
    }

    private IReadOnlyList<Step> BuildSteps(ColorScheme colors) =>
    [
        BuildStep(0, "Account", "Choose an account", "Account settings are ready.", colors),
        BuildStep(1, "Details", "Review preferences", "Notification and sync preferences.", colors),
        BuildStep(2, "Confirm", "Finish setup", "Everything is ready to submit.", colors),
    ];

    private Step BuildStep(int index, string title, string subtitle, string content, ColorScheme colors) => new(
        title: new Text(title),
        subtitle: new Text(subtitle),
        label: new Text((index + 1).ToString()),
        content: new Container(
            padding: new Thickness(12),
            color: Color.Parse("#FFF3EDF7"),
            child: new Text(content, fontSize: 14)),
        state: index < _currentStep
            ? StepState.Complete
            : index == _currentStep ? StepState.Editing : StepState.Indexed,
        isActive: index <= _currentStep,
        stepStyle: index == 2
            ? new StepStyle(
                Border: Plumix.Rendering.Border.FromBorderSide(new BorderSide(colors.Outline, 1)),
                Gradient: new LinearGradient([colors.SecondaryContainer, colors.PrimaryContainer]))
            : null);
}
